modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java

```diff
@@ -131,6 +133,27 @@ public Redirect getRedirect(
 
 		String layoutFriendlyURL = null;
 
+		Redirect skippedRedirect = null;
+
+		boolean skipRedirect = false;
+
+		String refererURL = httpServletRequest.getHeader(HttpHeaders.REFERER);
+
+		if (Validator.isNotNull(refererURL)) {
+			int questionPos = refererURL.indexOf(CharPool.QUESTION);
+
+			if (questionPos != -1) {
+				refererURL = refererURL.substring(0, pos);
+			}
+
+			if (refererURL.contains(
+					VirtualLayoutConstants.CANONICAL_URL_SEPARATOR +
+						GroupConstants.CONTROL_PANEL_FRIENDLY_URL)) {
+
+				skipRedirect = true;
+			}
+		}
+
```

Comment:
Can you refactor this code, splitting it into two smaller methods?
```
String referrerURL = _getReferrerURL(httpServletRequest);
String skipRedirect = _isSkipRedirect(referrerURL);
...
private String getReferrerURL(HttpServetRequest ...) {
		String refererURL = httpServletRequest.getHeader(HttpHeaders.REFERER);

                if (Validator.isNull(referrerURL)) {
                       return null;
                }

		int questionPos = refererURL.indexOf(CharPool.QUESTION);

		if (questionPos != -1) {
			return refererURL.substring(0, pos);
		}

		return refererURL;
}
```
Similarly with _isSkipRedirect().

Reply:
Hi @adolfopa
Would it be better to have `_isSkipRedirect` have no arguments and internally call `_getRefererURL`? This way we delegate all the relevant logic (getting the referer and checking if it's a control panel page) to `_isSkipRedirect`.

Reply:
@adolfopa , `refererURL` was introduced in this pr and it's not used anywhere else so that's not a problem